src/AbstractSQLRules2SQL.ts

```diff
@@ -1238,9 +1238,13 @@ const typeRules: Dictionary<MatchFn> = {
 			throw new SyntaxError('Any not supported on: ' + engine);
 		}
 		const value = AnyValue(getAbstractSqlQuery(args, 0), indent);
-		const innerType =
-			sbvrTypes[args[1] as keyof typeof sbvrTypes].types[engine];
-		return `ANY(CAST(${value} AS ${innerType}[]))`;
+		const castType = args[1] as keyof typeof sbvrTypes;
```

Comment:
If https://github.com/balena-io-modules/odata-to-abstract-sql/pull/163/files#r1922431368 is properly addressed this might become unnecessary.